src/app/shared/services/latestSearchesUtil.js

```diff
@@ -1,33 +1,34 @@
 export default class LatestSearchesUtil {
-    constructor(latestSearches) {
-        this.latestSearches = latestSearches;
-    }
-
     getLatestSearches() {
-        return this.latestSearches = JSON.parse(localStorage.getItem('latestSearches')) || [];
+        return JSON.parse(localStorage.getItem('latestSearches')) || [];
     };
 
-    addCity(city) {
+    addCity(city, cities) {
         if (!city || city.trim() === '') {
             return false;
         }
 
-        city = city.toLowerCase();
-        if (this.latestSearches.includes(city)) {
-            const index = this.latestSearches.indexOf(city);
+        city = city.trim().toLowerCase();
+        if (cities.includes(city)) {
+            const index = cities.indexOf(city);
             if (index > -1) {
-                this.latestSearches.splice(index, 1);
+                cities.splice(index, 1);
             }
         }
 
-        this.latestSearches.unshift(city);
-        if (this.latestSearches.length > 12) {
-            this.latestSearches.splice(12, this.latestSearches.length - 12);
+        if (cities.includes(city)) {
+            let index = cities.indexOf(city);
+            if (index > -1) {
+                cities.splice(index, 1);
+            }
```

Comment:
you are doing this twice, Is there a reason for this?

Reply:
Solved, I think that was from a bad update from development